Sources/TuistCache/Mappers/CacheGraphMutator.swift

```diff
@@ -123,31 +125,41 @@ class CacheGraphMutator: CacheGraphMutating {
 
     // swiftlint:disable:next function_body_length
     fileprivate func mapDependencies(
-        _ dependencies: Set<GraphDependency>,
-        graph: Graph,
+        for dependency: GraphDependency,
+        graphTraverser: GraphTraverser,
         graphDependencies: inout [GraphDependency: Set<GraphDependency>],
         precompiledArtifacts: [GraphTarget: AbsolutePath],
         sources: Set<String>,
         sourceTargets: inout Set<GraphTarget>,
         visitedPrecompiledArtifactPaths: inout [GraphTarget: VisitedArtifact?],
         loadedPrecompiledArtifacts: inout [AbsolutePath: GraphDependency]
     ) throws -> Set<GraphDependency> {
+        // Dependency already visited
+        guard graphDependencies[dependency] == nil else { return graphDependencies[dependency]! }
+
+        let dependencies = graphTraverser.dependencies[dependency] ?? Set()
         var newDependencies: Set<GraphDependency> = Set()
         try dependencies.forEach { dependency in
-            let graphTraverser = GraphTraverser(graph: graph)
-            let targetDependency: GraphTarget
-            switch dependency {
-            case let .target(name: name, path: path):
-                guard
-                    let target = graphTraverser.target(path: path, name: name)
-                else { return }
-                targetDependency = target
-            // If the dependency is not a target node we keep it.
-            default:
+            guard
+                case let GraphDependency.target(name: name, path: path) = dependency,
+                let targetDependency = graphTraverser.target(path: path, name: name)
+            else {
+                // If the dependency is not a target node we keep it as is.
                 newDependencies.insert(dependency)
                 return
             }
 
+            let mappedDependencies = try mapDependencies(
```

Comment:
Here you do it before the guard because you need to do it anyway for the dependencies of the target? Maybe I would add a comment to explain